packages/vitest/src/node/cli/cac.ts

```diff
@@ -0,0 +1,208 @@
+  let command = option.shorthand ? `-${option.shorthand}, --${commandName}` : `--${commandName}`
+  if ('argument' in option)
+    command += ` ${option.argument}`
+
+  function transform(value: unknown) {
+    if (!option.array && Array.isArray(value))
+      throw new Error(`Expected a single value for option "${command}"`)
+    if (option.transform)
+      return option.transform(value)
+    if (option.array)
+      return toArray(value)
+    if (option.normalize)
+      return normalize(String(value))
+    return value
+  }
+
+  if (option.description) {
+    cli.option(command, option.description, {
+      type: transform,
+    })
+  }
+
+  if ('subcommands' in option && option.subcommands) {
+    for (const commandName in option.subcommands) {
+      const subcommand = option.subcommands[commandName]
+      if (subcommand)
+        addCommand(cli, `${name}.${commandName}`, subcommand)
+    }
+  }
+}
+
+export function createCLI() {
+  const cli = cac('vitest')
+
+  cli
+    .version(version)
+
+  for (const optionName in cliOptionsConfig) {
+    const option = (cliOptionsConfig as any)[optionName] as CLIOption<any> | null
+    if (option)
+      addCommand(cli, optionName, option)
+  }
+
+  cli.help((info) => {
+    const options = info.find(current => current.title === 'Options')
+    if (typeof options !== 'object')
+      return info
+
+    const helpIndex = process.argv.findIndex(arg => arg === '--help')
+    const subcommands = process.argv.slice(helpIndex + 1)
+
+    const defaultOutput = options.body
+      .split('\n')
+      .filter(line => /^\s+--\S+\./.test(line) === false)
+      .join('\n')
+
+    // Filter out options with dot-notation if --help is not called with a subcommand (default behavior)
+    if (subcommands.length === 0) {
+      options.body = defaultOutput
+      return info
+    }
+
+    if (subcommands.length === 1 && (subcommands[0] === '--expand-help' || subcommands[0] === '--expandHelp'))
+      return info
+
+    const subcommandMarker = '$SUB_COMMAND_MARKER$'
+
+    const banner = info.find(current => /^vitest\/[0-9]+\.[0-9]+\.[0-9]+$/.test(current.body))
+    function addBannerWarning(warning: string) {
+      if (typeof banner?.body === 'string') {
+        if (banner?.body.includes(warning))
+          return
+
+        banner.body = `${banner.body}\n WARN: ${warning}`
+      }
+    }
+
+    // If other subcommand combinations are used, only show options for the subcommand
+    for (let i = 0; i < subcommands.length; i++) {
+      const subcommand = subcommands[i]
+
+      // --help --expand-help can't be called with multiple subcommands and is handled above
+      if (subcommand === '--expand-help' || subcommand === '--expandHelp') {
+        addBannerWarning('--expand-help subcommand ignored because, when used with --help, it must be the only subcommand')
+        continue
+      }
+
+      // Mark the help section for the subcommands
+      if (subcommand.startsWith('--')) {
```

Comment:
Are these two documented anywhere? Maybe at the end of `--help` it should print some instructions for this. 

Right now I had to read the source code before I understood why `vitest --coverage --help` didn't work.

Reply:
I actually have no idea what this code does 😄 , it's migrated from #3983

Reply:
Oh right it's actually something I've written: https://github.com/vitest-dev/vitest/pull/3983#issuecomment-1694595988

I'll check how we could add instructions for this into the `--help` message.

Reply:
Ok, I will wait for your feedback regarding this before merging